### ISSUE TITLE
Filter: Add "Media-attached" filter in combination with keywords

### ISSUE THREAD

**Issue by Bloodaxe95** [mastodon/mastodon#21062]
### Pitch

TL;DR: Keyword + media attached = filtered

I love the filter function in Mastodon, and it's an amazing and powerful feature. Allow us to filter content based on keywords AND if the content contains media. Some content I'm OK with being talked about, but I don't want it.... visualized, if you get me?

Example: Let's say I have arachnophobia. I don't mind people talking about spiders, but I do mind if there is an image that potentially contains spiders attached to it. So, all posts containing the word "spider" don't have to get filtered out, but the posts that cointain the word spider + an attached image should be filtered.

### Motivation

It would benefit us all by giving us even more powerful filtering options, better tailored to our needs and wishes.
